app/components/agenda/agendaitem/agendaitem-decision.js

```diff
@@ -73,31 +74,33 @@ export default class AgendaitemDecisionComponent extends Component {
     });
     await piece.save();
     this.args.treatment.set('report', piece);
-    this.args.treatment.save();
+    await this.args.treatment.save();
     this.isAddingReport = false;
     await this.loadReport.perform();
   }
 
-  @task
-  *attachNewReportVersion(piece) {
-    yield piece.save();
+  @action
+  async attachNewReportVersion(piece) {
+    await piece.save();
     this.args.treatment.set('report', piece);
-    yield this.args.treatment.save();
-    yield this.loadReport.perform();
+    await this.args.treatment.save();
+    await this.loadReport.perform();
   }
 
-  @task
-  *attachPreviousReportVersion(piece) {
-    // TODO: Assess if we need to go over container. `previousVersion` (if existant) might suffice?
-    const container = yield piece.documentContainer;
-    let remainingVersions = yield container.pieces;
-    if (remainingVersions.length) {
-      remainingVersions = remainingVersions.toArray();
-      const sortedPieces = sortPieces(remainingVersions);
-      this.args.treatment.set('report', sortedPieces[0]);
-      yield this.args.treatment.save();
+  @action
+  // eslint-disable-next-line no-unused-vars
+  async attachPreviousReportVersion(piece) {
+    /*
+      the deleted piece comes in from params, but not used here because we tracked the previous version.
+      Note: this used to be a task but something kept going wrong.
+      Trying to get piece.documentContainer after piece was destroyed kept failing after an ember upgrade:
+      "Cannot create a new tag for '<(unknown):ember890>' after it has been destroyed"
+    */
+    if (this.previousReport) {
+      this.args.treatment.set('report', this.previousReport);
```

Comment:
Since agenda-item-treatment data model has been converted to a native ES6 class, there is no need to use `set` anymore. Occurs a couple of times in this file. Not blocking for this PR.